clowder-framework/clowder2-demos
# CSV Concatenator

This adds an extractor that will try to concatenate CSV/TSV/XLSX files when uploaded to Clowder.

To test:
```
export CLOWDER_VERSION=2
python concatenate.py
```
(there's also a Dockerfile)

Then upload two CSVs to a dataset and see that concatenated.csv is created. If you already have 2+ CSVs in the dataset and you upload a new one, all will be merged into the output. Uploading additional CSVs will update the concatenated file as it goes.

File types are currently separated, meaning CSVs will get their own merge separate from Excel etc. This would be easy to change but not sure if there's a good use case for that?
![image](https://github.com/clowder-framework/clowder2-demos/assets/14166971/3b5e7472-893e-41dd-9d79-178443cdb392)

Uses pyclowder files.delete so requires this version of pyclowder: https://github.com/clowder-framework/pyclowder/pull/92
